src/System.Console/tests/TermInfo.cs

```diff
@@ -0,0 +1,83 @@
+    {
+        foreach (string location in ConsolePal.TermInfo.Database._terminfoLocations)
+        {
+            if (!Directory.Exists(location))
+                continue;
+
+            foreach (string term in Directory.EnumerateFiles(location, "*", SearchOption.AllDirectories))
+            {
+                if (term.ToUpper().Contains("README")) continue;
+                ConsolePal.TerminalColorInfo info = new ConsolePal.TerminalColorInfo(ConsolePal.TermInfo.Database.ReadDatabase(Path.GetFileName(term)));
+
+                if (!string.IsNullOrEmpty(info.ForegroundFormat))
+                {
+                    Assert.NotEmpty(ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.ForegroundFormat, 0 /* irrelevant, just an integer to put into the formatting*/));
+                }
+
+                if (!string.IsNullOrEmpty(info.BackgroundFormat))
+                {
+                    Assert.NotEmpty(ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.ForegroundFormat, 0 /* irrelevant, just an integer to put into the formatting*/));
+                }
+            }
+        }
+    }
+
+    [Theory]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00331m", "\u001B\u005B\u00341m", 1)]
+    [InlineData("screen", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("screen", "\u001B\u005B\u00332m", "\u001B\u005B\u00342m", 2)]
+    [InlineData("Eterm", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("Eterm", "\u001B\u005B\u00333m", "\u001B\u005B\u00343m", 3)]
+    [InlineData("wsvt25", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("wsvt25", "\u001B\u005B\u00334m", "\u001B\u005B\u00344m", 4)]
+    [InlineData("mach-color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("mach-color", "\u001B\u005B\u00335m", "\u001B\u005B\u00345m", 5)]
+    public void TermInfoVerification(string termToTest, string expectedForeground, string expectedBackground, int colorValue)
+    {
+        ConsolePal.TerminalColorInfo info = new ConsolePal.TerminalColorInfo(ConsolePal.TermInfo.Database.ReadDatabase(termToTest));
+        Assert.Equal(expectedForeground, ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.ForegroundFormat, colorValue));
+        Assert.Equal(expectedBackground, ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.BackgroundFormat, colorValue));
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.OSX)]
+    public void EmuTermInfoDoesntBreakParser()
+    {
+        // This file (available by default on OS X) is called out specifically since it contains a format where it has %i
+        // but only one variable instead of two. Make sure we don't break in this case
+        ConsolePal.TerminalColorInfo info = new ConsolePal.TerminalColorInfo(ConsolePal.TermInfo.Database.ReadDatabase("emu"));
```

Comment:
Rather than having this code, could you jut call TermInfoVerification("emu", ...)?  If we had the ability to have a PlatformSpecific InlineData, seems like this would be that, but since we don't, we're adding a fact that's platform-specific and then delegating.